File: leetcodeJS/array/1365.how-many-numbers-are-smaller-than-the-current-number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function (nums) {
  if (nums == [] || nums.length == 1) return nums;
  let p1 = 0;
  let p2 = 1;
  let solution = [];
  let cnt = 0;
  while (p1 < nums.length) {
    if (nums[p1] > nums[p2]) cnt++;
    if (p2 < nums.length) p2++;
    else {
      solution.push(cnt);
      cnt = 0;
      p1++;
      p2 = 0;
    }
  }
  return solution;
};

File: leetcodePython/array/1365.how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        if len(nums) == 0 or not nums:
            return
        solution = []
        for i in nums:
            cnt = 0
            for j in nums:
                if i > j:
                    cnt += 1

            solution.append(cnt)

        return solution
